lib/route/link/bridge_info.c

```diff
@@ -239,6 +256,55 @@ static struct rtnl_link_info_ops bridge_info_ops = {
 		}                                                                        \
 	} while (0)
 
+static int bridge_info_set_boolopt(struct rtnl_link *link,
+				   enum br_boolopt_id opt, bool value)
+{
+	struct bridge_info *bi = bridge_info(link);
+	uint32_t mask;
+
+	IS_BRIDGE_INFO_ASSERT(link);
+
+	if (opt < 0 || opt >= BR_BOOLOPT_MAX)
+		return -NLE_INVAL;
+
+	mask = 1ul << opt;
+
+	if (value)
+		bi->b_boolopts.optval |= mask;
+	else
+		bi->b_boolopts.optval &= ~mask;
+
+	bi->b_boolopts.optmask |= mask;
+	bi->ce_mask |= BRIDGE_ATTR_BOOLOPT;
```

Comment:
There is no API to unset `bi->ce_mask & BRIDGE_ATTR_BOOLOPT` or `bi->b_boolopts.optmask & mask`.

That's a common lack in many setters, so it's probably fine. 

If we make `value` an `int`, we could take "-1" to unset `bi->b_boolopts.optmask`. Not sure that is useful though. It's also slightly confusing. So maybe not.

Fine-as-is, unless you'd like to change it.

Reply:
I also noticed the common lack of a way to unset values. I'm not sure of the usefulness of that though, since you could just not set the value in the first place. Maybe it would be useful for getting an object, modifying it a bit, and sending it back as a change request which matches a different object. That seems like an odd use case though. I think I'll leave it as is for now. If there is a desire in the future for ways to unset values, maybe a `reset` or `clear` version of the functions could be added.